examples-gallery/plot_sliding_block.py

```diff
@@ -0,0 +1,306 @@
+
+# %%
+""""
+sliding a block on a road
+=========================
+A block, modeled as a particle is sliding on a road to cross a
+hill. The block is subject to gravity and speed dependent
+friction. 
+Gravity points in the negative Y direction.
+A force tangential to the road is applied to the block.
+Three objective functions to me minimized may be selected:
+
+- selektion = 0: time to reach the end point is minimized
+- selektion = 1: time to reach the end point is minimized and 
+  the energy consumed is minimized.
+- selektion = 2: energy consumed is minimized.
+
+**Constants**
+
+- m: mass of the block [kg]
+- g: acceleration due to gravity [m/s**2]
+- reibung: coefficient of friction [N/(m*s)]
+ -a, b: paramenters determining the shape of the road.
+
+**States**
+
+- x: position of the block [m]
+- ux: velocity of the block [m/s]
+
+**Specifieds**
+
+- F: force applied to the block [N]
+
+"""
+import sympy.physics.mechanics as me
+from collections import OrderedDict
+import numpy as np
+import sympy as sm
+from opty.direct_collocation import Problem
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+
+# %%
+# select the objective function to be minimized
+selektion = 2
```

Comment:
For an example it would be better if you just work out both solutions and show them both. The website isn't interactive so the person would have to download the example and run the alternative to see the results.

Reply:
> For an example it would be better if you just work out both solutions and show them both. The website isn't interactive so the person would have to download the example and run the alternative to see the results.

I did not know this.
It's actually three different objective functions. Let me see whether I can do all three from a time point of view